File: Interviewbit/Tree/right-view-of-binary-tree.cpp
/**
 * Definition for binary tree
 * struct TreeNode {
 *     int val;
 *     TreeNode *left;
 *     TreeNode *right;
 *     TreeNode(int x) : val(x), left(NULL), right(NULL) {}
 * };
 */
vector<int> Solution::solve(TreeNode* A) {
    if(A==NULL) return {};
    queue<TreeNode*> q;
    q.push(A);
    TreeNode* tmp;
    vector<int> ans;
    while(!q.empty()){
        int size=q.size();
        
        for(int i=1;i<=size;i++){
            tmp=q.front();
            q.pop();
            if(i==size){
                ans.push_back(tmp->val);
            }
            if(tmp->left) q.push(tmp->left);
            if(tmp->right) q.push(tmp->right);
        }
    }
    return ans;
}

File: Interviewbit/Tree/right-view-of-binary-tree.java
/**
 * Definition for binary tree
 * class TreeNode {
 *     int val;
 *     TreeNode left;
 *     TreeNode right;
 *     TreeNode(int x) {
 *      val = x;
 *      left=null;
 *      right=null;
 *     }
 * }
 */
public class Solution {
    public ArrayList<Integer> solve(TreeNode A) {
        if(A==null) return new ArrayList<>();
        
        TreeNode tmp;
        Queue<TreeNode> q=new LinkedList<>();
        q.add(A);
        ArrayList<Integer> ans=new ArrayList<>();
        
        while(!q.isEmpty()){
            int size=q.size();
            for(int i=1;i<=size;i++){
                tmp=q.poll();
                if(i==size){
                    ans.add(tmp.val);
                }
                if(tmp.left!=null) q.add(tmp.left);
                if(tmp.right!=null) q.add(tmp.right);
            }
        }
        return ans;
    }
}
